integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForDeleteCommand.scala

```diff
@@ -104,24 +111,26 @@ private[sql] case class CarbonProjectForDeleteCommand(
       // handle the clean up of IUD.
       CarbonUpdateUtil.cleanUpDeltaFiles(carbonTable, false)
 
-      val deletedSegments = DeleteExecution.deleteDeltaExecution(
+      val (deletedSegments, deletedRowCountTemp) = DeleteExecution.deleteDeltaExecution(
         databaseNameOp,
         tableName,
         sparkSession,
         dataRdd,
         timestamp,
         isUpdateOperation = false,
         executorErrors)
+
+      if (executorErrors.failureCauses != FailureCauses.NONE) {
```

Comment:
I think it better check whether it fail before execute horizontal compaction, right?